File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.cpp
#include <string>
#include <vector>
#include <map>
#include <algorithm>
using namespace std;

int solution(int k, vector<int> tangerine) {
    int answer = 0;
        map<int, int> cnt;
    vector<int> cnt2;
    for (auto t : tangerine)
    {
        cnt[t] += 1;
	}
    
    for (auto c : cnt)
    {
        cnt2.push_back(c.second);
    }

    sort(cnt2.begin(), cnt2.end());
    for (auto it = cnt2.rbegin();it != cnt2.rend();it++)
    {
        k -= *it;
        answer++;
        if (k <= 0) break;
    }
    
    
    return answer;
}

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.cs
using System;
using System.Linq;
using System.Collections.Generic;
public class Solution {
    public int solution(int k, int[] tangerine) {
        int answer = 0;
        int cnt = 0;
        Dictionary<int,int> dict = new Dictionary<int,int>();

        for (int i = 0; i < tangerine.Length; i++)
        {
            if (dict.ContainsKey(tangerine[i]))
            {
                dict[tangerine[i]]++;
            }
            else
            {
                dict[tangerine[i]] = 1;
            }
        }
        var sortedListAsc = dict.OrderByDescending(item => item.Value).ToList();

        foreach (var item in sortedListAsc)
        {
            cnt += item.Value;
            answer++;
            if (k <= cnt) break;
        }
        return answer;
    }
}